src/perf/lib/RpsClient.h

```diff
@@ -24,6 +24,13 @@ class RpsClient : public PerfBase {
     }
 
     ~RpsClient() override {
+        if (Connections != nullptr) {
```

Comment:
Now that I look at this, I guess Connections could be an array of `ConnectionScope` and then you don't have to do anything manually, right?